pkg/volume/gcepd/attacher.go

```diff
@@ -384,9 +384,29 @@ func (detacher *gcePersistentDiskDetacher) Detach(volumeName string, nodeName ty
 }
 
 func (detacher *gcePersistentDiskDetacher) UnmountDevice(deviceMountPath string) error {
+	if runtime.GOOS == "windows" {
+		// Flush data cache for windows because it does not do so automatically during unmount device
+		exec := detacher.host.GetExec(gcePersistentDiskPluginName)
+		err := writeVolumeCache(deviceMountPath, exec)
+		if err != nil {
+			klog.Errorf("writeVolumeCache (windows) failed with error %s", err)
+		}
+		return err
+	}
 	return mount.CleanupMountPoint(deviceMountPath, detacher.host.GetMounter(gcePersistentDiskPluginName), false)
 }
 
+
+func writeVolumeCache(deviceMountPath string, exec mount.Exec) error {
+	cmd := fmt.Sprintf("get-volume -filePath %s | write-volumecache", deviceMountPath)
+	output, err := exec.Run("powershell", "/c", cmd)
+	if err != nil {
+		err = errors.New(err.Error() + " " + string(output))
```

Comment:
Does output or error contain information about what command was run?

Reply:
it should contain, but I added into error message explicitly to make sure